From 078b214af05c1a95f26e288d7b6547a8f7f369d0 Mon Sep 17 00:00:00 2001
From: Jerome Forissier <jerome.forissier@linaro.org>
Date: Thu, 18 Oct 2018 11:16:23 +0200
Subject: [PATCH] qemu: increase CFG_DTB_MAX_SIZE to 1 MiB

Since upstream QEMU commit 14ec3cbd7c1e ("device_tree: Increase
FDT_MAX_SIZE to 1 MiB"), which is included in release v2.12.1 and later,
OP-TEE initialization fails with the following error (-3 is
-FDT_ERR_NOSPACE):

 E/TC:0 0 init_fdt:808 Invalid Device Tree at 0x40000000: error -3

Increase CFG_DTB_MAX_SIZE accordingly. Tested with the current tip of the
QEMU master branch, in 32- and 64-bit modes (note that our 64-bit QEMU
setup needs a TF-A patch -- PLAT_QEMU_DT_MAX_SIZE needs to be set to 1 MiB
too).

Signed-off-by: Jerome Forissier <jerome.forissier@linaro.org>
Acked-by: Jens Wiklander <jens.wiklander@linaro.org>
---
 core/arch/arm/plat-vexpress/conf.mk | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/core/arch/arm/plat-vexpress/conf.mk b/core/arch/arm/plat-vexpress/conf.mk
index 91442801b1..7535c03834 100644
--- a/core/arch/arm/plat-vexpress/conf.mk
+++ b/core/arch/arm/plat-vexpress/conf.mk
@@ -90,6 +90,7 @@ endif
 $(call force,CFG_BOOT_SECONDARY_REQUEST,y)
 $(call force,CFG_PSCI_ARM32,y)
 $(call force,CFG_DT,y)
+CFG_DTB_MAX_SIZE ?= 0x100000
 # SE API is only supported by QEMU Virt platform
 CFG_SE_API ?= y
 CFG_SE_API_SELF_TEST ?= y
@@ -108,4 +109,5 @@ CFG_SHMEM_SIZE  ?= 0x00200000
 # When Secure Data Path is enable, last MByte of TZDRAM is SDP test memory.
 CFG_TEE_SDP_MEM_SIZE ?= 0x00400000
 $(call force,CFG_DT,y)
+CFG_DTB_MAX_SIZE ?= 0x100000
 endif
